File: Data Structures & Algorithms/find-duplicate-integer/submission-6.cpp
class Solution {
public:
    int findDuplicate(vector<int>& nums) {
        int elem = 0;
        while (true){
            if (nums[elem] == 0) return elem;
            int tmp = nums[elem];
            nums[elem] = 0;
            elem = tmp;
        }
    }
};

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-6.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        map = {value: idx for idx, value in enumerate(nums)}
        res = 0
        for num in nums:
            if num - 1 not in map.keys():
                cnt = 0
                tmp = num
                while tmp in map.keys():
                    cnt += 1
                    tmp += 1
                res = max(res, cnt)

        return res